Cython/Compiler/PyrexTypes.py

```diff
@@ -702,6 +708,11 @@ def __init__(self, base_dtype, axes):
         if not self.dtype.is_fused:
             self.dtype_name = Buffer.mangle_dtype_name(self.dtype)
 
+    def make_dont_clear_temps_type(self):
+        new_type = copy.copy(self)
+        new_type.clear_temps_after_use = False
+        return new_type
+
```

Comment:
The flag doesn't really feel like a part of the type information, does it? Can't we make the `additional_hashing` flag of temps semantically meaningful and store the information there?